test/upsert/mzcompose.py

```diff
@@ -308,3 +309,34 @@ def workflow_incident_49(c: Composition) -> None:
             c.run("testdrive", "incident-49/02-after-rehydration.td")
 
         c.run("testdrive", "incident-49/03-reset.td")
+
+
+def workflow_rocksdb_cleanup(c: Composition) -> None:
+    """Testing rocksdb cleanup after dropping sources"""
+    c.down(destroy_volumes=True)
+    dependencies = [
+        "materialized",
+        "zookeeper",
+        "kafka",
+        "schema-registry",
+    ]
+    c.up(*dependencies)
+
+    testdrive_files = [
+        "drop-source.td",
+        "drop-cluster-cascade.td",
+        "drop-source-in-cluster.td",
+    ]
+
+    for testdrive_file in testdrive_files:
+        c.run("testdrive", f"rocksdb-cleanup/{testdrive_file}")
+        files = c.exec(
+            "materialized",
+            "bash",
+            "-c",
+            "find /mzdata/source_data -type f",
+            capture=True,
+        ).stdout.strip()
+        assert (
+            files == ""
+        ), f"The scratch directory should have been cleaned up but has following files:\n{files}"
```

Comment:
instead of this, you should check that the `<cluster_id>-cluster-<replica_id>` directory of the clusters you dropped are gone, right? you can use `exec()` with `psql` (like in the `ssh-connection` mzcompose) to get the cluster names

Reply:
`c.sql_query` please, not exec'ing `psql`!

Reply:
Oh cool, let me do that. I was wondering how to get the cluster and replica IDs to show that some source with data as well.

Reply:
Made changes to get the cluster level and source level rocksdb paths and checking them for presence of files instead of the top level scratch directory.